libbpfgo.go

```diff
@@ -121,6 +122,22 @@ int bpf_prog_detach_cgroup_legacy(
 	return syscall(__NR_bpf, BPF_PROG_DETACH, &attr, sizeof(attr));
 }
 
+struct bpf_link * bpf_prog_attach_iter(struct bpf_program *prog, __u32 map_fd)
```

Comment:
@grantseltzer WDYT of naming internal functions with a prefix or suffix avoiding confusion with the libbpf ones?

Reply:
The `union bpf_iter_link_info` added a new field recently, should we need to support it?
https://github.com/libbpf/libbpf/commit/981001bf468ec5bf4bd9bcbdd378f5748b515c0c

Reply:
@mozillazg Do you suspect that this was the source of the previous error message? I'm thinking perhaps that field existed in a linked object but not in a header (or vice versa?)

Reply:
I don't fully understand what this field does but yes it would be useful to support it.